async-query-core/src/main/java/org/opensearch/sql/spark/dispatcher/model/FlintIndexOptions.java

```diff
@@ -33,6 +34,11 @@ public boolean autoRefresh() {
     return Boolean.parseBoolean(getOption(AUTO_REFRESH).orElse("false"));
   }
 
+  public boolean isExternalScheduler() {
+    // Default is false, which means using internal scheduler to refresh the index.
```

Comment:
Internal/External sounds ambiguous to me. Can we rephrase or add documentation?

Reply:
It's documented here https://github.com/opensearch-project/opensearch-spark/blob/main/docs/index.md#flint-index-refresh

Reply:
Does internal means within spark? I think it is difficult to find that documentation from this option defined in SQL plugin. And I would imagine `internal` is internal to OpenSearch...

Reply:
yeah internal means within `spark` and external could be anything.

Reply:
I don't want to create a blocker now, but @ykmr1224  raised a good question. 

We should have the name of this flag from the perspective of spark. Probably `use_spark_scheduler = True | False` would make more sense because we actually don't care about what an external scheduler is?

Reply:
it's to allow extensibility in case of behavior difference around external schedulers